Firestore/core/src/firebase/firestore/local/local_store.cc

```diff
@@ -377,17 +382,42 @@ bool LocalStore::ShouldPersistQueryData(const QueryData& new_query_data,
   return changes > 0;
 }
 
+absl::optional<QueryData> LocalStore::GetQueryData(const core::Query& query) {
+  auto target_id = target_id_by_query_.find(query);
+  if (target_id != target_id_by_query_.end()) {
+    return query_data_by_target_[target_id->second];
+  }
+  return query_cache_->GetTarget(query);
+}
+
 void LocalStore::NotifyLocalViewChanges(
     const std::vector<local::LocalViewChanges>& view_changes) {
   persistence_->Run("NotifyLocalViewChanges", [&] {
     for (const LocalViewChanges& view_change : view_changes) {
+      int target_id = view_change.target_id();
+
       for (const DocumentKey& key : view_change.removed_keys()) {
         persistence_->reference_delegate()->RemoveReference(key);
       }
-      local_view_references_.AddReferences(view_change.added_keys(),
-                                           view_change.target_id());
+      local_view_references_.AddReferences(view_change.added_keys(), target_id);
       local_view_references_.RemoveReferences(view_change.removed_keys(),
-                                              view_change.target_id());
+                                              target_id);
+
+      if (!view_change.is_from_cache()) {
+        const auto& query_data = query_data_by_target_.find(target_id);
```

Comment:
It's actually `query_data->second` that's the query data. Consider naming this `entry` or `found` and then creating a `const QueryData& query_data = entry->second;` after the `HARD_ASSERT` to more clearly distinguish the entry containing key and value from just the value.

Reply:
Changed it to use `entry` for the iterator, and added a second variable for `query_data`.